src/catalog/src/durable/impls/persist.rs

```diff
@@ -453,6 +456,39 @@ impl PersistHandle {
             .sorted_by(|a, b| Ord::cmp(&b.ts, &a.ts))
     }
 
+    /// Generates an iterator of [`StateUpdate`] that contain all updates to the catalog
+    /// state up to, but not including, `upper`.
+    async fn snapshot_unconsolidated(
+        &mut self,
+        upper: Timestamp,
+    ) -> impl Iterator<Item = StateUpdate> + DoubleEndedIterator {
+        let snapshot = if upper > 0.into() {
```

Comment:
nit: I got thrown by the `0.into()` when reading this through the first time. Could we `Timestamp::minimum()` instead?

Reply:
Done